File: src/main/java/org/gd/leetcode/p0392/Solution.java
package org.gd.leetcode.p0392;

import org.gd.leetcode.common.LeetCode;

/**
 * https://leetcode.com/problems/is-subsequence/
 */
@LeetCode(
        name = "Is Subsequence",
        difficulty = LeetCode.Level.EASY,
        state = LeetCode.State.DONE,
        tags = {
                LeetCode.Tags.DYNAMIC_PROGRAMMING,
                LeetCode.Tags.BINARY_SEARCH,
                LeetCode.Tags.GREEDY
        }
)
class Solution {

    public boolean isSubsequence(String subSeq, String seq) {
        final int subSeqLength = subSeq.length();
        int i = 0;
        for (int j = 0, seqLength = seq.length(); i < subSeqLength && j < seqLength; j++) {
            if (subSeq.charAt(i) == seq.charAt(j))
                i++;
        }
        return i == subSeqLength;
    }
}

File: src/main/java/org/gd/leetcode/p0392/solution.py
class Solution:
    def isSubsequence(self, sub_seq: str, seq: str) -> bool:

        sub_seq_len: int = len(sub_seq)
        i: int = 0

        seq_len = len(seq)
        j: int = 0

        while i < sub_seq_len and j < seq_len:
            if sub_seq[i] == seq[j]:
                i += 1
            j += 1

        return i == sub_seq_len
